README.md

```diff
@@ -1,155 +1,168 @@
+# mooc-grader
+
 This grading service accepts anonymous submissions for grading via HTTP. The
 grading can be done synchronously or asynchronously using a submission queue.
 The results are delivered to the calling system and the grader does not keep
 any record other than service logs. The grader is designed to serve exercises
 for the A+ learning system. A farm of individual grading servers can be setup
 to handle large amount of submissions.
 
-The grader is implemented on Django 1.9 (`grader/settings.py`). The application 
-is tested on both Python 2.7 and 3.4. 
+The grader is implemented on Django 1.9 (`grader/settings.py`). The application
+is tested on both Python 2.7 and 3.4.
 
 The grader can be run stand alone without the full stack to test graders in
 the local system environment. The grader is designed to be extended for
 different courses and exercise types. Course and exercise configuration is in
 `courses` directory where further documentation and examples are available.
 
-Installing for development
-==========================
+## Installing for development
 
 > 6/2014 - Ubuntu 12.04.4
 
-1. ### Clone the software
-
-	General requirements
+### 1. Clone the software
 
-		sudo apt-get install git libjpeg-dev
-		sudo apt-get install libxml2-dev libxslt-dev zlib1g-dev
+General requirements
 
-	Install software
+    sudo apt-get install git libjpeg-dev
+    sudo apt-get install libxml2-dev libxslt-dev zlib1g-dev
 
-		git clone https://github.com/Aalto-LeTech/mooc-grader.git
-		mkdir mooc-grader/uploads
+Install software
 
-2. ### Python requirements (2.7 should work too)
+    git clone https://github.com/Aalto-LeTech/mooc-grader.git
+    mkdir mooc-grader/uploads
 
-		sudo apt-get install python3 python3-dev python3-pip
+### 2. Python requirements (2.7 should work too)
 
-	Make sure pip and virtualenv are installed.
+    sudo apt-get install python3 python3-dev python3-pip python3-venv
 
-		sudo pip3 install virtualenv
 
-	Then, create virtual environment with grader requirements.
+Then, create virtual environment with grader requirements.
 
-		virtualenv -p python3 venv
-		source venv/bin/activate
-		pip install -r mooc-grader/requirements.txt
+    python3 -m venv venv
+    source venv/bin/activate
```

Comment:
Side note: someone once told me that `source` and `.` would behave differently. To my understanding, they are the same and `source` is an alias in Bash whereas `.` is the standard name (in POSIX?). What do you think?

Reply:
The [Bash manual](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html) describes `source` as "A synonym for .". To my knowledge the behaviour is identical